remix/app/root.tsx

```diff
@@ -8,17 +11,45 @@ import {
   Scripts,
   ScrollRestoration,
   isRouteErrorResponse,
+  useLoaderData,
   useRouteError,
 } from '@remix-run/react';
 import { StatusCodes } from 'http-status-codes';
 
+import { commitSession, getSession } from './config/session.server';
+import makeToast from './lib/toast/makeToast';
 import stylesheet from '../app/stylesheets/tailwind.css';
 
 export const links: LinksFunction = () => [
   { rel: 'stylesheet', href: stylesheet },
 ];
 
+export async function loader({ request }: LoaderArgs) {
+  const session = await getSession(request.headers.get('Cookie'));
+  const toastMessage = session.get('toastMessage') || null;
+  const toastMode = session.get('toastMode') || null;
+
+  return json(
+    { toastMessage, toastMode },
+    {
+      headers: {
+        'Set-Cookie': await commitSession(session),
+      },
+    }
+  );
+}
+
 export default function App() {
+  const { toastMessage, toastMode } = useLoaderData<typeof loader>();
+
+  useEffect(() => {
+    if (!toastMessage) {
+      return;
+    }
+
+    makeToast(toastMode, toastMessage);
+  }, [toastMessage]);
```

Comment:
It might be worth adding [this article as reference](https://www.jacobparis.com/content/remix-form-toast) in the insights section of the pull request as using cookie storage for such use case is unexpected 😅 I am wondering if using localStorage might be better than cookie storage 💡

Reply:
Yep, for that I just forgot to do added the reference that 😅, using `localStorage` might be a better choice here, as we can keep the cookie more secure. Thank you for mentioning that 🙏🏼.

I will improve.